namui/src/namui/system/file/local_storage/write.rs

```diff
@@ -0,0 +1,60 @@
+use super::get_root_directory::get_root_directory;
+use crate::{file::types::PathLike, simple_error_impl};
+use js_sys::Uint8Array;
+use std::path::PathBuf;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::JsFuture;
+
+const CHUNK_SIZE: usize = 4 * 1024; // 4 KiB
+
+#[derive(Debug)]
+pub enum WriteError {
+    InvalidFileName(String),
+    PathShouldBeAbsolute(String),
+    Other(String),
+}
+simple_error_impl!(WriteError);
+
+pub async fn write(path_like: impl PathLike, content: impl AsRef<[u8]>) -> Result<(), WriteError> {
+    let file_path = path_like.path();
+    if !file_path.has_root() {
+        return Err(WriteError::PathShouldBeAbsolute(format!("{file_path:?}")));
+    }
+    let file_name = match file_path.file_name() {
+        Some(file_name) => file_name.to_string_lossy().to_string(),
+        None => return Err(WriteError::InvalidFileName(format!("{file_path:?}"))),
+    };
+    let parent_directory_path = match file_path.parent().as_deref() {
+        Some(path) => path.to_path_buf(),
+        None => PathBuf::new(),
+    };
+    let root = get_root_directory().await?;
+    let parent_directory_handle = root
+        .get_directory_handle_recursively(
+            parent_directory_path,
+            crate::file::local_storage::file_system_handle::GetHandleOption { create: true },
+        )
+        .await?;
+    let file_handle = parent_directory_handle
+        .get_file_handle(
+            file_name,
+            super::file_system_handle::GetHandleOption { create: true },
+        )
+        .await?;
+
+    let file_stream = file_handle.create_writable().await?;
+    let writer = file_stream.get_writer()?;
+    for chunk in content.as_ref().chunks(CHUNK_SIZE) {
+        let unit8array = Uint8Array::new_with_length(chunk.len() as u32);
+        unit8array.copy_from(chunk);
+        JsFuture::from(writer.write_with_chunk(&unit8array)).await?;
+    }
+    JsFuture::from(writer.close()).await?;
+    Ok(())
```

Comment:
I think this code will have a performance trouble by bottle neck. Did you test the bps?
This code waits until stream write a chunk. If stream is fill with chunks, it will stop this for loop, and when stream is ready to get data after consume, we push a new chunk, so at that time, we have a delay between 'stream ready(in js)' and 'push data(wasm)', including js-wasm context switch.

First of all, do we need to chunk the data before write the data? This function gets parameter `content` which have whole data. `write_with_chunk` with chunk should be used when the source of data is stream type. But it's not a stream right now.

So, I suggest two options.
1. write whole binary data at once
2. make a stream type and get content as stream and pipe that to file's writable stream.

I think the first option is good for start. the second will be highest performance because it use full stream from source to storage. But we don't have stream yet in namui system, so we can extends this function with stream later when we make a stream type.

Reply:
This is a good article to understand what i'm saying, https://iximiuz.com/en/posts/nodejs-readable-streams-distilled/

Reply:
Take a look at this to make zero-copy https://github.com/rustwasm/wasm-bindgen/issues/2456

Reply:
Good. As you said, there was suuuper bottleneck.
![image](https://user-images.githubusercontent.com/38313680/205246172-24fd3b69-58cc-42b6-8641-18dc345c2cf4.png)
It was tested with only 1MiB.

## Changes
- Do not split chunks.
- Do not copy data.